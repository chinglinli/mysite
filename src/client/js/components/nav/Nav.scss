@import '../../../css/base.scss';

.root {
  background: $dominant-color2;
  position: relative;
  height: 45px;

  .leftMenu {
    padding-left: 10px;
  }

  .rightMenu {
    position: absolute;
    top: 0;
    right: 10px;
    text-align: right;
  }

  .menuItem {
    @extend .link;
    display: inline-block;
    padding: 0 15px;
    line-height: 45px;
    font-size: 1.2em;
    color: white;
    font-weight: 700;

    &:hover {
      background: $accent-color;
    }
  }
}