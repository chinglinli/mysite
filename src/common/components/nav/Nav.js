import React from 'react'
import PropTypes from 'prop-types'
import ImmutablePropTypes from 'react-immutable-proptypes'
import { Link } from 'react-router'

import styles from './Nav.scss'

export default class Nav extends React.PureComponent {
  static propTypes = {
    leftMenuItems: ImmutablePropTypes.listOf(
      ImmutablePropTypes.contains({
        url: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired
      })
    ),
    rightMenuItems: ImmutablePropTypes.listOf(
      ImmutablePropTypes.contains({
        url: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired
      })
    )
  }

  render () {
    return (
      <div className={styles.root}>
        <div className={styles.menus}>
          {this.props.leftMenuItems.map(item => (
            <Link key={item.get('url')} to={item.get('url')} className={styles.menuItem}>{item.get('name')}</Link>
          ))}
        </div>
        <div className={styles.splitter}/>
        <div className={styles.menus}>
          {this.props.rightMenuItems.map(item => (
            <Link key={item.get('url')} to={item.get('url')} className={styles.menuItem}>{item.get('name')}</Link>
          ))}
        </div>
      </div>
    )
  }
}
