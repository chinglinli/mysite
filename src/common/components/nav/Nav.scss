@import '../../css/base.scss';

.root {
  display: flex;
  background: $dominant-color2;
  height: 2.8125rem;

  .menus {
    margin: 0 1rem;
  }

  .splitter {
    flex-grow: 1;
  }

  .menuItem {
    @extend .link;
    display: inline-block;
    padding: 0 1rem;
    line-height: 2.8125rem;
    font-size: 1.2em;
    color: white;
    font-weight: 700;

    &:hover {
      background: $accent-color;
    }
  }
}