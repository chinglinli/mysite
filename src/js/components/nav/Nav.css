@value dominant-color2 from '../../css/colors.css';

.root {
    background: dominant-color2;
}

.left {

}

.right {

}
