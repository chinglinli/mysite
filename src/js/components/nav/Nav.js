import * as React from 'react';

import styles from './Nav.css';


export default class Nav extends React.Component {

    render() {
        const leftMenuItems = [
            'Python', '專案作品', '程式設計', 'UNIX & 工具', '胡言亂語'
        ];

        const rightMenuItems = [
            '閱讀計劃', '成功日記'
        ];

        return (
            <div className={styles.root}>
                <ul className={styles.left}>
                    {leftMenuItems.map(item => (<li><a href=''>{item}</a></li>))}
                </ul>
                <ul className={styles.root}>
                    {rightMenuItems.map(item => (<li><a href=''>{item}</a></li>))}
                </ul>
            </div>
        )
    }
}
