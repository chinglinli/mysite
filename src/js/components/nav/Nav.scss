@import '../../../css/colors.scss';

.root {
    background: $dominant-color2;
    position: relative;
    height: 45px;

    .leftMenu {
        display: inline-block;
        margin-left: 20px;
    }

    .rightMenu {
        position: absolute;
        display: inline-block;
        right: 20px;
    }

    .menuItem {
        display: inline-block;
        padding: 0 15px;
        line-height: 45px;
        font-size: 1.2em;
        color: white;

        &:hover {
            background: $accent-color;
        }
    }
}