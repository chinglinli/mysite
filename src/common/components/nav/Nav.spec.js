import Immutable from 'immutable'
import React from 'react'
import renderer from 'react-test-renderer'

import Nav from './Nav'


test('It should render Nav component correctly', () => {
  const props = Immutable.fromJS({
    leftMenuItems: [
      {url: 'url 1', name: 'name 1'},
      {url: 'url 2', name: 'name 2'},
      {url: 'url 3', name: 'name 3'},
    ],
    rightMenuItems: [
      {url: 'url 4', name: 'name 4'},
      {url: 'url 5', name: 'name 5'},
      {url: 'url 6', name: 'name 6'},
    ]
  })

  const component = renderer.create(
    <Nav
      leftMenuItems={props.get('leftMenuItems')}
      rightMenuItems={props.get('rightMenuItems')}
    />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})