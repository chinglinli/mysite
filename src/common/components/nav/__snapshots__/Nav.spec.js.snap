// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render Nav component correctly 1`] = `
<div
  className="root"
>
  <div
    className="menus"
  >
    <a
      className="menuItem"
      onClick={[Function]}
      style={Object {}}
    >
      name 1
    </a>
    <a
      className="menuItem"
      onClick={[Function]}
      style={Object {}}
    >
      name 2
    </a>
    <a
      className="menuItem"
      onClick={[Function]}
      style={Object {}}
    >
      name 3
    </a>
  </div>
  <div
    className="splitter"
  />
  <div
    className="menus"
  >
    <a
      className="menuItem"
      onClick={[Function]}
      style={Object {}}
    >
      name 4
    </a>
    <a
      className="menuItem"
      onClick={[Function]}
      style={Object {}}
    >
      name 5
    </a>
    <a
      className="menuItem"
      onClick={[Function]}
      style={Object {}}
    >
      name 6
    </a>
  </div>
</div>
`;
